game-core/src/main/java/org/triplea/performance/PerfTimer.java

```diff
@@ -23,52 +24,88 @@
 @Log
 public class PerfTimer implements Closeable {
 
-  private static final PerfTimer DISABLED_TIMER = new PerfTimer("disabled");
   private static final Map<String, AtomicLong> runningTotal = new HashMap<>();
-  final String title;
-  private final long startMillis;
+  private static final Map<String, AtomicLong> runningCount = new HashMap<>();
+  private final String title;
+  private final Optional<Long> startNanos;
+  private final int reportingFrequency;
 
-  private PerfTimer(final String title) {
+  private PerfTimer(final String title, final int reportingFrequency) {
     this.title = title;
-    this.startMillis = System.nanoTime();
+    this.reportingFrequency = reportingFrequency;
+    // If reporting frequency is 0, avoid the overhead of querying the time.
+    if (this.reportingFrequency > 0) {
+      this.startNanos = Optional.of(System.nanoTime());
+    } else {
+      this.startNanos = Optional.empty();
+    }
   }
 
   private long stopTimer() {
-    return System.nanoTime() - startMillis;
+    return System.nanoTime() - startNanos.get();
```

Comment:
I'm concerned about the `stopTimer` method. My thoughts:
- the `get()` method of optional generally should be avoided. This is a pretty good article on optional usage: https://dzone.com/articles/using-optional-correctly-is-not-optional, it sums up a number of similar points.

I think one to help this is to mark `startNanos` as @Nullable and just use a null value and do a classic `if( null ) ` type of check.


Though, we still have a lot of internal state going that route. Even if we allow for disabled timers, assigning startNanos always to `System.nanoTime` would not hurt that much (we are still talking a `ns` operation, that is not a place to make performance improvements).

Reply:
Removed the optional stuff for now.